src/app/pages/system/alert-service/alert-service-list/alert-service-list.component.ts

```diff
@@ -16,93 +24,129 @@ import { WebSocketService } from 'app/services/ws.service';
 @UntilDestroy()
 @Component({
   selector: 'ix-alert-service-list',
-  template: '<ix-entity-table [title]="title" [conf]="this"></ix-entity-table>',
+  templateUrl: './alert-service-list.component.html',
+  styleUrls: ['./alert-service-list.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AlertServiceListComponent implements EntityTableConfig<AlertService> {
-  title = 'Alert Services';
-  routeAddTooltip = this.translate.instant('Add Alert Service');
-  queryCall = 'alertservice.query' as const;
-  wsDelete = 'alertservice.delete' as const;
-  protected routeSuccess: string[] = ['system', 'alertservice'];
-  routeAdd: string[] = ['system', 'alertservice', 'add'];
-  routeEdit: string[] = ['system', 'alertservice', 'edit'];
-  entityTable: EntityTableComponent<AlertService>;
-  actionsOutOfHeader = true;
+export class AlertServiceListComponent implements OnInit {
+  readonly requiresRoles = [Role.AlertListWrite];
+  readonly providerList = alertServiceNames.map((alertService) => alertService.value);
 
-  columns = [
-    { name: 'Service Name', prop: 'name', always_display: true },
-    { name: 'Type', prop: 'type' },
-    { name: 'Level', prop: 'level' },
-    { name: 'Enabled', prop: 'enabled', selectable: true },
-  ];
-  config = {
-    paging: true,
-    sorting: { columns: this.columns },
-    deleteMsg: {
-      title: 'Alert Service',
-      key_props: ['name'],
-    },
-  };
+  dataProvider: AsyncDataProvider<AlertService>;
+  filterString = '';
+
+  columns = createTable<AlertService>([
+    textColumn({
+      title: this.translate.instant('Service Name'),
+      propertyName: 'name',
+    }),
+    textColumn({
+      title: this.translate.instant('Type'),
+      propertyName: 'type',
+      getValue: (service) => this.translate.instant(
+        alertServiceNames.find((alertService) => alertService.value === service.type).label,
+      ),
+    }),
+    textColumn({
+      title: this.translate.instant('Level'),
+      propertyName: 'level',
+      getValue: (service) => this.translate.instant(alertLevelLabels.get(service.level)),
+    }),
+    textColumn({
+      title: this.translate.instant('Enabled'),
+      propertyName: 'enabled',
+      getValue: (service) => (service.enabled ? this.translate.instant('Yes') : this.translate.instant('No')),
+    }),
+    actionsColumn({
+      cssClass: 'wide-actions',
+      actions: [
+        {
+          iconName: 'edit',
+          tooltip: this.translate.instant('Edit'),
+          onClick: (row) => this.editAlertService(row),
+          hidden: (row) => of(!this.providerList.includes(row.type)),
```

Comment:
What does this check do? Are there alert services we want to hide?

Reply:
@undsoft - it's some logic from the old code, gonna remove it 👍